### PR TITLE
Adding signed typed data

### PR DESCRIPTION
**Pull request by Valazan**
Adding the signed Typed data to the DappKit

### CODE REVIEW

**Review by davidecarpini — COMMENTED** [review]
LGTM added some comments to check

File: packages/dapp-kit/src/classes/certificate-wallet.ts [hunk 1]
```diff
@@ -60,6 +62,15 @@ class CertificateBasedWallet implements ConnexWallet {
         options: Connex.Signer.TxOptions,
     ): Promise<Connex.Vendor.TxResponse> => this.wallet.signTx(msg, options);
 
+    signTypedData = (
+        _domain: ethers.TypedDataDomain,
+        _types: Record<string, ethers.TypedDataField[]>,
+        _value: Record<string, unknown>,
+        _options?: SignTypedDataOptions,
+    ): Promise<string> => {
+        return this.wallet.signTypedData(_domain, _types, _value);
```
[the remark this answers is not in the document]
Comment (Valazan) [reply packages/dapp-kit/src/classes/certificate-wallet.ts:71]: was a bug, options should be added

[the remark this answers is not in the document]
Comment (Valazan) [reply packages/dapp-kit/src/classes/certificate-wallet.ts:71]: The work is done by the wallet manager which has a catch logic:
```

    signTypedData = (
        domain: ethers.TypedDataDomain,
        types: Record<string, ethers.TypedDataField[]>,
        value: Record<string, unknown>,
        options?: SignTypedDataOptions,
    ): Promise<string> =>
        this.wallet.signTypedData(domain, types, value, options).catch((e) => {
            DAppKitLogger.error('WalletManager', 'signTypedData', e);
            throw e;
        });
```

File: packages/dapp-kit/src/classes/wc-wallet.ts [hunk 2]
```diff
@@ -23,6 +25,16 @@ class WCWallet implements ConnexWallet {
...
     ): Promise<Connex.Vendor.TxResponse> => this.signer.signTx(msg, options);
 
+    signTypedData = async (
+        _domain: ethers.TypedDataDomain,
+        _types: Record<string, ethers.TypedDataField[]>,
+        _value: Record<string, unknown>,
+        _options?: SignTypedDataOptions,
+    ): Promise<string> => {
+        // Delegating the signTypedData to the signer instance
+        return this.signer.signTypedData(_domain, _types, _value, _options);
```
[the remark this answers is not in the document]
Comment (Valazan) [reply packages/dapp-kit/src/classes/wc-wallet.ts:35]: We are following the similar implementation of WC on signTx and signMessage so I assume it should be ok

**Review by davidecarpini — APPROVED** [review]

**Pull request merged**